### PR TITLE
CODENVY-363: add error messages when user reach the RAM limits

### PR DESCRIPTION
**Pull request by olexii4**
Signed-off-by: Oleksii Orel <oorel@codenvy.com>
@ashumilova, @evoevodin

### CODE REVIEW

File: core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java [hunk 1]
```diff
@@ -235,14 +242,40 @@ void checkMaxEnvironmentRam(WorkspaceConfig config) throws LimitExceededExceptio
...
                                                         "This value is set by your admin with the 'limits.workspace.env.ram' property",
-                                                        maxRamPerEnv,
-                                                        workspaceRam));
+                                                        parseMegabyteSizeToString(maxRamPerEnv, "mb"),
+                                                        parseMegabyteSizeToString(workspaceRam, "mb")));
             }
         }
     }
 
+    private String parseMegabyteSizeToString(long ramSizeMB, String suffix) {
```
Comment (sleshchenko) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java]: Maybe it is better to use https://github.com/eclipse/che/blob/master/core/commons/che-core-commons-lang/src/main/java/org/eclipse/che/commons/lang/Size.java#L22-22 here.

**Comment by ashumilova** [comment]
+1 for dashboard

**Comment by sleshchenko** [comment]
:+1:

File: core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java (hunk 1 above)
Comment (evoevodin) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java]: Well, about the method: i don't think this is a good place for such kind of transformations. 

1. Basically you need this `parse` method  only in the one place `checkRamAndPropagateStart`
2. You don't need to parse from megabytes to megabytes.
3. If [Size]( https://github.com/eclipse/che/blob/master/core/commons/che-core-commons-lang/src/main/java/org/eclipse/che/commons/lang/Size.java) class doesn't work appropriate for you, then you should either simplify this method to convert megabytes to gigabytes(and that's it), or add a new method into the `Size` class, which may be reused by others.

File: core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java [hunk 2]
```diff
@@ -235,14 +242,40 @@ void checkMaxEnvironmentRam(WorkspaceConfig config) throws LimitExceededExceptio
                                                  .mapToInt(machineCfg -> machineCfg.getLimits().getRam())
                                                  .sum();
             if (workspaceRam > maxRamPerEnv) {
-                throw new LimitExceededException(format("The maximum RAM per workspace is set to '%dmb' and you requested '%dmb'. " +
+                throw new LimitExceededException(format("The maximum RAM per workspace is set to '%s' and you requested '%s'. " +
                                                         "This value is set by your admin with the 'limits.workspace.env.ram' property",
-                                                        maxRamPerEnv,
-                                                        workspaceRam));
+                                                        parseMegabyteSizeToString(maxRamPerEnv, "mb"),
```
Comment (evoevodin) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java]: Can you please explain why did you change this code?
I think the previous version was pretty ok, and we don't need extra overhead here.

File: core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java [hunk 3]
```diff
@@ -169,21 +171,26 @@ public WorkspaceImpl updateWorkspace(String id, Workspace update) throws Conflic
...
-                                                                            .getEnvironment(ws.getRuntime().getActiveEnv())
-                                                                            .get()
-                                                                            .getMachineConfigs())
-                                                               .mapToLong(this::sumRam)
-                                                               .sum();
+            final List<WorkspaceImpl> workspacesPerUser = getWorkspaces(user);
+            final long runningWorkspaces = workspacesPerUser.stream().filter(ws -> RUNNING == ws.getStatus()).count();
+            final long currentlyUsedRamMB = workspacesPerUser.stream().filter(ws -> STOPPED != ws.getStatus())
+                                                             .map(ws -> ws.getConfig()
+                                                                          .getEnvironment(
```
Comment (evoevodin) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java]: change the formatting back please `.getEnvironment(ws.getRuntime().getActiveEnv())`

File: core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java [hunk 4]
```diff
@@ -169,21 +171,26 @@ public WorkspaceImpl updateWorkspace(String id, Workspace update) throws Conflic
...
-            final long currentlyUsedRamMB = getWorkspaces(user).stream()
-                                                               .filter(ws -> STOPPED != ws.getStatus())
-                                                               .map(ws -> ws.getConfig()
-                                                                            .getEnvironment(ws.getRuntime().getActiveEnv())
-                                                                            .get()
-                                                                            .getMachineConfigs())
-                                                               .mapToLong(this::sumRam)
-                                                               .sum();
+            final List<WorkspaceImpl> workspacesPerUser = getWorkspaces(user);
+            final long runningWorkspaces = workspacesPerUser.stream().filter(ws -> RUNNING == ws.getStatus()).count();
```
Comment (evoevodin) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java]: Workspace uses RAM when it in `STARTING`, `RUNNING`, `STOPPING` state so you must change the code `RUNNING  = ws.getStatus()` to `STOPPED != ws.getStatus()`

File: core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java [hunk 5]
```diff
@@ -56,6 +57,8 @@
 @Singleton
 public class LimitsCheckingWorkspaceManager extends WorkspaceManager {
 
+    public static final DecimalFormat DECIMAL_FORMAT = new DecimalFormat("#0.#");
```
Comment (evoevodin) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java]: why `public` ?

**Comment by evoevodin** [comment]
+1
